### PR TITLE
(S/L) Problem: We jsonify JSON string unnecessarily.

### PR DESCRIPTION
**Pull request by jana-rapava**
Solution: Add a check for this case.

Signed-off-by: Jana Rapava <janarapava@eaton.com>

### CODE REVIEW

File: src/fty_common_utf8.cc [hunk 1]
```diff
@@ -329,6 +329,14 @@ s_jsonify_translation_string (const char *key, va_list args)
 {
     log_trace ("called");
 
+    // check if 'key' is already in JSON format
+    // since we may start/end with {{vari}} and we don't want to go through the whole string,
+    // check just first two and last two characters
+    size_t length = strlen (key);
+    if (key[0] == '{' && key[1] != '{' && key[length-2] != '}' && key[length-1] == '}') {
```
Comment (EldoreiJK) [inline src/fty_common_utf8.cc]: just for the sanity check if length is at least 2

Comment (jimklimov) [reply src/fty_common_utf8.cc]: good point, I missed ;)

Comment (jana-rapava) [reply src/fty_common_utf8.cc]: Good catch!

File: src/fty_common_utf8.cc [hunk 2]
```diff
@@ -329,6 +329,14 @@ s_jsonify_translation_string (const char *key, va_list args)
 {
     log_trace ("called");
 
+    // check if 'key' is already in JSON format
+    // since we may start/end with {{vari}} and we don't want to go through the whole string,
+    // check just first two and last two characters
+    size_t length = strlen (key);
+    if (length >= 2 && key[0] == '{' && key[1] != '{' && key[length-2] != '}' && key[length-1] == '}') {
```
Comment (jimklimov) [inline src/fty_common_utf8.cc]: Technically, `{{}}` is already `length==4` ;)

Comment (EldoreiJK) [reply src/fty_common_utf8.cc]: yeah, but the worst case is that it will check some insane case, but won't crash due to invalid indexes.

**Pull request merged**